server/pulp/server/db/model/__init__.py

```diff
@@ -667,6 +665,20 @@ def pre_save_signal(cls, sender, document, **kwargs):
         if not document._storage_path:
             document._storage_path = FileStorage.get_path(document)
 
+    def set_file(self, name):
+        """
+        Update the storage path to include the specified file name.
+        This should only be called when the content unit is associated with
+        a single file.
+
+        :param name: A file name.
+        :type name: str
+        """
+        if self.storage_path:
+            return
+        path = FileStorage.get_path(self)
+        self._storage_path = os.path.join(path, name)
```

Comment:
Is self.storage_path a property that returns self._storage_path? If so, can we make that property also be the setter so this doesn't need to access _storage_path directly? It might be a little cleaner and less confusing.